### PR TITLE
platform/nuttx: Support riscv based chips

### PR DESCRIPTION
**Pull request by no1wudi**

### CODE REVIEW

File: product-mini/platforms/nuttx/wamr.mk [hunk 1]
```diff
@@ -62,6 +66,35 @@ else ifeq (${WAMR_BUILD_TARGET}, XTENSA)
   CFLAGS += -DBUILD_TARGET_XTENSA
   INVOKE_NATIVE := invokeNative_xtensa.s
   AOT_RELOC := aot_reloc_xtensa.c
+else ifeq (${WAMR_BUILD_TARGET}, RISCV64)
+
+ifeq (${CONFIG_ARCH_FPU},y)
+  CFLAGS += -DBUILD_TARGET_RISCV64_LP64F
+  INVOKE_NATIVE += invokeNative_riscv64_lp64d.s
```
Comment (wenyongh) [inline product-mini/platforms/nuttx/wamr.mk:73]: LP64F isn't supported yet, not sure whether using invokeNative_riscv64_lp64d.s has potential issues?

File: product-mini/platforms/nuttx/wamr.mk [hunk 2]
```diff
@@ -62,6 +66,35 @@ else ifeq (${WAMR_BUILD_TARGET}, XTENSA)
...
+  CFLAGS += -DBUILD_TARGET_RISCV64_LP64
+  INVOKE_NATIVE += invokeNative_riscv64_lp64.s
+endif
+
+  AOT_RELOC :=
+
+else ifeq (${WAMR_BUILD_TARGET}, RISCV32)
+
+ifeq (${CONFIG_ARCH_FPU}, y)
+  CFLAGS += -DBUILD_TARGET_RISCV64_ILP32F
```
Comment (wenyongh) [inline product-mini/platforms/nuttx/wamr.mk:87]: Same as above, no asm src file is added here. Do you want to implement the related codes to support LP64F and ILP32F?

File: product-mini/platforms/nuttx/wamr.mk (hunk 1 above)
Comment (no1wudi) [reply product-mini/platforms/nuttx/wamr.mk]: Removed,

File: product-mini/platforms/nuttx/wamr.mk (hunk 2 above)
Comment (no1wudi) [reply product-mini/platforms/nuttx/wamr.mk]: Yes, and AOT support should also be considered

File: product-mini/platforms/nuttx/wamr.mk [hunk 3]
```diff
@@ -62,6 +66,34 @@ else ifeq (${WAMR_BUILD_TARGET}, XTENSA)
   CFLAGS += -DBUILD_TARGET_XTENSA
   INVOKE_NATIVE := invokeNative_xtensa.s
   AOT_RELOC := aot_reloc_xtensa.c
+else ifeq (${WAMR_BUILD_TARGET}, RISCV64)
+
+ifeq (${CONFIG_ARCH_FPU},y)
+  $(error riscv64 lp64f don't support)
```
Comment (wenyongh) [inline product-mini/platforms/nuttx/wamr.mk:72]: Had better change to isn't supported or is unsupported?

File: product-mini/platforms/nuttx/wamr.mk [hunk 4]
```diff
@@ -62,6 +66,34 @@ else ifeq (${WAMR_BUILD_TARGET}, XTENSA)
...
+  CFLAGS += -DBUILD_TARGET_RISCV64_LP64
+  INVOKE_NATIVE += invokeNative_riscv64_lp64.s
+endif
+
+  AOT_RELOC :=
+
+else ifeq (${WAMR_BUILD_TARGET}, RISCV32)
+
+ifeq (${CONFIG_ARCH_FPU}, y)
+  $(error riscv32 ilp32f don't support)
```
Comment (wenyongh) [inline product-mini/platforms/nuttx/wamr.mk:86]: Same as above

File: product-mini/platforms/nuttx/wamr.mk (hunk 2 above)
Comment (wenyongh) [reply product-mini/platforms/nuttx/wamr.mk]: Great! Do you need any help from us?

**Comment by no1wudi** [comment]
@wenyongh I had open another PR about generic riscv support, we can discuss here #649

**Pull request merged**